### PR TITLE
Fixes #1497 - Do not query app versions in MarathonHealthCheckManager

### LINKED ISSUE #1497: CPU usage increased dramatically 0.8.1-RC1 -> master
I built and deployed 964e43035830bc48c56e4853e39ed44553383427. Running against 0.22.1 masters:

image

Upgrade started at 16:20, last node was updated at 16:31. Revert to 0.8.1-RC1 happened at 16:41.

I mentioned performance in #1472 as well.

### PR DESCRIPTION
**Pull request by kolloch**
.statuses(appId: PathId)

and make MarathonHealthCheckManager data structures more efficient

### CODE REVIEW

File: src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala [hunk 1]
```diff
@@ -28,14 +28,11 @@ class MarathonHealthCheckManager @Inject() (
...
 
-  // composite key for partitioning the set of active health checks
-  protected[this] case class AppVersion(id: PathId, version: Timestamp)
-
   protected[this] case class ActiveHealthCheck(
     healthCheck: HealthCheck,
     actor: ActorRef)
 
-  protected[this] var appHealthChecks = Map[AppVersion, Set[ActiveHealthCheck]]()
+  protected[this] var appHealthChecks: Map[PathId, Map[Timestamp, Set[ActiveHealthCheck]]] = Map.empty
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala]: Create a map with default Value, so you can get rid of all subsequent getOrlElse statements.

File: src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala [hunk 2]
```diff
@@ -197,25 +207,20 @@ class MarathonHealthCheckManager @Inject() (
     }
 
   override def statuses(appId: PathId): Future[Map[String, Seq[Health]]] = withReadLock {
-    val futureVersions = appRepository.listVersions(appId)
     implicit val timeout: Timeout = Timeout(2, SECONDS)
+    val futureHealths = for {
+      ActiveHealthCheck(_, actor) <- appHealthChecks.getOrElse(appId, Map.empty).values.iterator.flatten.toVector
```
Comment (aquamatthias) [inline src/main/scala/mesosphere/marathon/health/MarathonHealthCheckManager.scala]: additional iterator needed?

**Comment by aquamatthias** [comment]
Cool! Especially since the statuses() performance is greatly improved.
Made a suggestions regarding default value - otherwise looks good.

**Comment by kolloch** [comment]
Followed up on your suggestion. Thanks!

**Comment by aquamatthias** [comment]
:+1:

**Pull request merged**